### PR TITLE
Defect fixes

### LINKED ISSUE #625: [BUG]: redfish_storage_volume failing to create Raid 1 volume starting from collection 8.3.0
### Bug Description

Hello
i have a simple playbook to create RAID1 volume with 2 disks. 
Process works fine with collection version 8.2.0. Failure starting with 8.3.0 and up

### Component or Module Name

redfish_storage_volume

### Ansible Version

Ansible 2.16.3

### Python Version

Python 3.11

### iDRAC/OME/OME-M version

idrac 2.85.85.85

### Operating System

N/A

### Playbook Used

```
- name: iDrac configuration 
  hosts: localhost
  gather_facts: false

  vars:
    lom_username: root
    lom_password: 
    alv_lom_ip: 
    p_controller: RAID.Integrated.1-1
    p_disks: [
      "Disk.Bay.0:Enclosure.Internal.0-1:RAID.Integrated.1-1",
      "Disk.Bay.1:Enclosure.Internal.0-1:RAID.Integrated.1-1",
      "Disk.Bay.2:Enclosure.Internal.0-1:RAID.Integrated.1-1"
      ]
  tasks:

    - name: Create v_disk using first two physical disks
      dellemc.openmanage.redfish_storage_volume:
        baseuri: "{{ lom_ip }}"
        username: "{{ lom_username }}"
        password: "{{ lom_password }}"
        validate_certs: false
        # volume_type: "Mirrored"
        raid_type: RAID1
        name: "VDISK1"
        controller_id: "{{ p_controller }}"
        drives: "{{ p_disks[:2] }}"
        state: "present"
```

### Logs
**when using collection 8.3.0**

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'SupportedRAIDTypes'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707435559.7119524-289348-30705886827210/AnsiballZ_redfish_storage_volume.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707435559.7119524-289348-30705886827210/AnsiballZ_redfish_storage_volume.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707435559.7119524-289348-30705886827210/AnsiballZ_redfish_storage_volume.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.dellemc.openmanage.plugins.modules.redfish_storage_volume', init_globals=dict(_module_fqn='ansible_collections.dellemc.openmanage.plugins.modules.redfish_storage_volume', _modlib_path=modlib_path),\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_0rc51nk7/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 711, in <module>\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_0rc51nk7/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 698, in main\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_0rc51nk7/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 624, in configure_raid_operation\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_0rc51nk7/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 550, in perform_volume_create_modify\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_0rc51nk7/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 537, in check_raid_type_supported\nKeyError: 'SupportedRAIDTypes'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

**when using collection 8.7.0**

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: '@Redfish.OperationApplyTimeSupport'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707434824.733956-289147-81803729536349/AnsiballZ_redfish_storage_volume.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707434824.733956-289147-81803729536349/AnsiballZ_redfish_storage_volume.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/z-marek.talas@ap.autoliv.int/.ansible/tmp/ansible-tmp-1707434824.733956-289147-81803729536349/AnsiballZ_redfish_storage_volume.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.dellemc.openmanage.plugins.modules.redfish_storage_volume', init_globals=dict(_module_fqn='ansible_collections.dellemc.openmanage.plugins.modules.redfish_storage_volume', _modlib_path=modlib_path),\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_27wdjagu/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 939, in <module>\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_27wdjagu/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 910, in main\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_27wdjagu/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 665, in check_apply_time_supported_and_reboot_required\n  File \"/tmp/ansible_dellemc.openmanage.redfish_storage_volume_payload_27wdjagu/ansible_dellemc.openmanage.redfish_storage_volume_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/redfish_storage_volume.py\", line 649, in get_apply_time\nKeyError: '@Redfish.OperationApplyTimeSupport'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

### Steps to Reproduce

run the playbook

### Expected Behavior

new vdisk gets created

### Actual Behavior

playbook error

### Screenshots

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ShivamSh3**
# Description
Following defects are fixed:
JIT-281593
JIT-281767
JIT-282006
JIT-282007
JIT-281758
JIT-287848

# GitHub Issues
https://github.com/dell/dellemc-openmanage-ansible-modules/issues/625

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below
```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/ome_application_console_preferences.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 7.0.0
-# Copyright (C) 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline plugins/modules/ome_application_console_preferences.py]:
```suggestion
# Copyright (C) 2022-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: plugins/modules/ome_device_local_access_configuration.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 8.1.0
-# Copyright (C) 2021-2023 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline plugins/modules/ome_device_local_access_configuration.py]:
```suggestion
# Copyright (C) 2021-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: plugins/modules/ome_devices.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 6.1.0
-# Copyright (C) 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline plugins/modules/ome_devices.py]:
```suggestion
# Copyright (C) 2022-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: plugins/modules/ome_devices.py [hunk 4]
```diff
@@ -420,6 +422,8 @@ def main():
     )
     try:
         with RestOME(module.params, req_session=True) as rest_obj:
+            if module.params.get("job_wait") and module.params.get("job_wait_timeout") <= 0:
```
Comment (sachin-apa) [inline plugins/modules/ome_devices.py]: This condition can be moved before creating the RESTOME object.

File: tests/unit/plugins/modules/test_ome_application_console_preferences.py [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 7.0.0
-# Copyright (C) 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline tests/unit/plugins/modules/test_ome_application_console_preferences.py]:
```suggestion
# Copyright (C) 2022-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: tests/unit/plugins/modules/test_ome_device_local_access_configuration.py [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 8.1.0
-# Copyright (C) 2021-2023 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline tests/unit/plugins/modules/test_ome_device_local_access_configuration.py]:
```suggestion
# Copyright (C) 2021-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: tests/unit/plugins/modules/test_ome_devices.py [hunk 7]
```diff
@@ -2,8 +2,8 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 6.1.0
-# Copyright (C) 2021-2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.1.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (sachin-apa) [inline tests/unit/plugins/modules/test_ome_devices.py]:
```suggestion
# Copyright (C) 2021-2024 Dell Inc. or its subsidiaries. All Rights Reserved.
```

File: plugins/modules/ome_application_console_preferences.py (hunk 1 above)
Comment (ShivamSh3) [reply plugins/modules/ome_application_console_preferences.py]: done

File: plugins/modules/ome_devices.py (hunk 3 above)
Comment (ShivamSh3) [reply plugins/modules/ome_devices.py]: done

File: tests/unit/plugins/modules/test_ome_application_console_preferences.py (hunk 5 above)
Comment (ShivamSh3) [reply tests/unit/plugins/modules/test_ome_application_console_preferences.py]: done

File: plugins/modules/ome_devices.py (hunk 4 above)
Comment (ShivamSh3) [reply plugins/modules/ome_devices.py]: done

File: tests/unit/plugins/modules/test_ome_devices.py (hunk 7 above)
Comment (ShivamSh3) [reply tests/unit/plugins/modules/test_ome_devices.py]: done

File: tests/unit/plugins/modules/test_ome_device_local_access_configuration.py (hunk 6 above)
Comment (ShivamSh3) [reply tests/unit/plugins/modules/test_ome_device_local_access_configuration.py]: done

File: plugins/modules/ome_device_local_access_configuration.py (hunk 2 above)
Comment (ShivamSh3) [reply plugins/modules/ome_device_local_access_configuration.py]: done

**Review by singh-lovepreet1 — APPROVED** [review]

**Review by rajshekarp87 — DISMISSED** [review]

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**